src/plugins/data/public/search/search_service.ts

```diff
@@ -174,12 +174,14 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
           this.sessionCache.set(dataFrame.name, dataFrame.meta.sessionId);
         }
         const existingIndexPattern = indexPatterns.getByTitle(dataFrame.name!, true);
-        const dataSet = await indexPatterns.create(
-          dataFrameToSpec(dataFrame, existingIndexPattern?.id),
-          !existingIndexPattern?.id
-        );
-        // save to cache by title because the id is not unique for temporary index pattern created
-        indexPatterns.saveToCache(dataSet.title, dataSet);
+        if (existingIndexPattern) {
```

Comment:
actually we want to create a dataset without an existing index pattern or not. the problem is that we try to get indiceswhen we create the temp index pattern even tho s3 is availabe

Reply:
we want to check the data source if it's like default or not